program_transformers/src/rollups/rollup_persister.rs

```diff
@@ -0,0 +1,571 @@
+            let Ok((rollup_to_verify, rollup)) = self.get_rollup_to_verify().await else {
+                continue;
+            };
+            let Some(rollup_to_verify) = rollup_to_verify else {
+                // no rollups to persist
+                continue;
+            };
+            let Ok(rollup) = rollup
+                .map(|r| bincode::deserialize::<Rollup>(r.rollup_binary_bincode.as_slice()))
+                .transpose()
+                .map_err(|e| ProgramTransformerError::DeserializationError(e.to_string()))
+            else {
+                continue;
+            };
+            self.persist_rollup(rollup_to_verify, rollup.map(Box::new))
+                .await;
+        }
+    }
+
+    pub async fn persist_rollup(
+        &self,
+        mut rollup_to_verify: rollup_to_verify::Model,
+        mut rollup: Option<Box<Rollup>>,
+    ) {
+        info!("Persisting {} rollup", &rollup_to_verify.url);
+        loop {
+            match &rollup_to_verify.rollup_persisting_state {
+                &RollupPersistingState::ReceivedTransaction => {
+                    if let Err(err) = self
+                        .download_rollup(&mut rollup_to_verify, &mut rollup)
+                        .await
+                    {
+                        error!("Error during rollup downloading: {}", err)
+                    };
+                }
+                &RollupPersistingState::SuccessfullyDownload => {
+                    if let Some(r) = &rollup {
+                        self.validate_rollup(&mut rollup_to_verify, r).await;
+                    } else {
+                        error!(
+                            "Trying to validate non downloaded rollup: {:#?}",
+                            &rollup_to_verify
+                        )
+                    }
+                }
+                &RollupPersistingState::SuccessfullyValidate => {
+                    if let Some(r) = &rollup {
+                        // TODO: Add retry?
+                        if let Err(e) = self.store_rollup_update(&mut rollup_to_verify, r).await {
+                            error!("Store rollup update: {}", e)
+                        };
+                    } else {
+                        error!(
+                            "Trying to store update for non downloaded rollup: {:#?}",
+                            &rollup_to_verify
+                        )
+                    }
+                }
+                &RollupPersistingState::FailedToPersist | &RollupPersistingState::StoredUpdate => {
+                    if let Err(e) = self.drop_rollup_from_queue(&rollup_to_verify).await {
+                        error!("failed to drop rollup from queue: {}", e);
+                    };
+                    info!(
+                        "Finish processing {} rollup file with {:?} state",
+                        &rollup_to_verify.url, &rollup_to_verify.rollup_persisting_state
+                    );
+                    return;
```

Comment:
This is just a note that probably when we bring this to the wider RPC community we might want to add some specific metrics based on their ops preferences as well.

Reply:
Thank you. I added some metrics but not sure if i use statsd correctly (in context of nft_ingester), so i will be very appreciate if you will look at that

Reply:
We can see if wider community has specific suggestions later when this PR goes to them, but as a reference implementation this looks good to me.